### PR TITLE
resolve: follow-up for synthesizing empty name

### LINKED ISSUE #22401: resolved: new handling of root domain looks wrong
This is mostly a tracking bug because I think #22231 isn't right. i.e. it should probably only synthesize NXDOMAIN for A/AAAA lookups in the root zone, but let all other root zone lookups through as is, so that DNSSEC can work.

But I didn't test this, this is just an issue I am pretty sure exists, based on looking at the change.

### PR DESCRIPTION
**Pull request by yuwata**
Follow-up for #22231.
Closes #22401.

Before this PR.
```
$ dig -t SRV '.'

; <<>> DiG 9.16.24-RH <<>> -t SRV .
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 16836
;; flags: qr aa rd ra ad; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;.				IN	SRV

;; Query time: 1 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Fri Feb 04 12:01:09 JST 2022
;; MSG SIZE  rcvd: 28
```

After this PR.
```
$ dig -t SRV '.'

; <<>> DiG 9.16.24-RH <<>> -t SRV .
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 19861
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;.				IN	SRV

;; AUTHORITY SECTION:
.			86394	IN	SOA	a.root-servers.net. nstld.verisign-grs.com. 2022020302 1800 900 604800 86400

;; Query time: 20 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Fri Feb 04 12:00:12 JST 2022
;; MSG SIZE  rcvd: 103
```

### CODE REVIEW

File: src/resolve/resolved-dns-query.c [hunk 1]
```diff
@@ -734,15 +734,11 @@ int dns_query_go(DnsQuery *q) {
...
                 DnsScopeMatch match;
-                const char *name;
 
-                name = dns_question_first_name(dns_query_question_for_protocol(q, s->protocol));
-                if (!name)
-                        continue;
-
-                match = dns_scope_good_domain(s, q->ifindex, q->flags, name);
+                match = dns_scope_good_domain(s, q);
                 if (match < 0) {
```
Comment (anitazha) [inline src/resolve/resolved-dns-query.c]: can match ever be `< 0`? looking at `dns_scope_good_domain()` it doesn't seem so

**Review by anitazha — APPROVED** [review]
Change LGTM w.r.t fixing the reported issue but I'm not confident enough to merge it.

File: src/resolve/resolved-dns-query.c (hunk 1 above)
Comment (yuwata) [reply src/resolve/resolved-dns-query.c]: Indeed. The conditions are dropped. Thanks.

File: src/resolve/resolved-dns-query.c [hunk 2]
```diff
@@ -770,18 +760,8 @@ int dns_query_go(DnsQuery *q) {
...
-                if (!name)
-                        continue;
-
-                match = dns_scope_good_domain(s, q->ifindex, q->flags, name);
-                if (match < 0) {
-                        log_debug("Couldn't check if '%s' matches against scope, ignoring.", name);
-                        continue;
-                }
 
+                match = dns_scope_good_domain(s, q);
```
Comment (poettering) [inline src/resolve/resolved-dns-query.c:768]: maybe add `assert(match >= 0);` here, encoding the assumption that this can never fail.

Comment (yuwata) [reply src/resolve/resolved-dns-query.c:768]: Sure. The assertion is added.

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**